packages/@tinacms/fields/src/ColorPicker/ColorPicker.tsx

```diff
@@ -140,82 +143,148 @@ interface State {
-  static defaultProps = {
-    colorFormat: ColorFormat.Hex,
-  }
-  presetColors = [
-    '#D0021B',
-    '#F5A623',
-    '#F8E71C',
-    '#8B572A',
-    '#7ED321',
-    '#417505',
-    '#BD10E0',
-    '#9013FE',
-    '#4A90E2',
-    '#50E3C2',
-    '#B8E986',
-    '#000000',
-    '#4A4A4A',
-    '#9B9B9B',
-    '#FFFFFF',
-  ]
-
-  state = {
-    displayColorPicker: false,
-  }
+const presetColors = [
+  '#D0021B',
+  '#F5A623',
+  '#F8E71C',
+  '#8B572A',
+  '#7ED321',
+  '#417505',
+  '#BD10E0',
+  '#9013FE',
+  '#4A90E2',
+  '#50E3C2',
+  '#B8E986',
+  '#000000',
+  '#4A4A4A',
+  '#9B9B9B',
+  '#FFFFFF',
+]
 
-  get colorFormat() {
-    return (this.props.colorFormat || ColorFormat.Hex).toLowerCase()
-  }
+export const ColorPicker:React.FC<Props> = ({
+    colorFormat,
+    userColors = presetColors,
+    widget = "sketch",
+    input
+}) => {
+  const [displayColorPicker, setDisplayColorPicker] = useState(false);
 
-  handleClick = () => {
-    this.setState({ displayColorPicker: !this.state.displayColorPicker })
-  }
+  const getColorFormat = (colorFormat || ColorFormat.Hex).toLowerCase();
+  const getColorRGBA = input.value ? ColorFormatter[getColorFormat].parse(input.value) : null
+  const handleClick = () => setDisplayColorPicker(!displayColorPicker);
+  const handleClose = () => setDisplayColorPicker(false);
 
-  handleClose = () => {
-    this.setState({ displayColorPicker: false })
-  }
-
-  handleChange = (pickerColor: any) => {
+  const handleChange = (pickerColor: any) => {
     const color = (pickerColor.hex === nullColor
       ? null
-      : { ...pickerColor.rgb, a: 1 }) as ColorRGBA | null
-    this.props.input.onChange(
-      color ? ColorFormatter[this.colorFormat].getValue(color) : null
+      : { ...pickerColor.rgb, a: 1}) as ColorRGBA | null
+    input.onChange(
+      color ? ColorFormatter[getColorFormat].getValue(color) : null
     )
   }
 
-  get colorRGBA() {
-    return this.props.input.value
-      ? ColorFormatter[this.colorFormat].parse(this.props.input.value)
-      : null
-  }
-
-  render() {
-    return (
-      <>
-        <Swatch
-          onClick={this.handleClick}
-          colorRGBA={this.colorRGBA}
-          colorFormat={this.colorFormat}
-        />
-        {this.state.displayColorPicker ? (
-          <>
-            <Popover>
-              <SketchPicker
-                presetColors={[...this.presetColors, nullColor]}
-                color={this.colorRGBA || { r: 0, g: 0, b: 0, a: 0 }}
-                onChange={this.handleChange}
-                disableAlpha={true}
-                width={'240px'}
-              />{' '}
-            </Popover>
-            <Cover onClick={this.handleClose} />
-          </>
-        ) : null}{' '}
-      </>
-    )
-  }
+  return (
+    <>
+      <Swatch
+        onClick={handleClick}
+        colorRGBA={getColorRGBA}
+        colorFormat={getColorFormat}
+      />
+      {displayColorPicker ? (
```

Comment:
There's a couple changes I'd like to see here.

First, pull the `SketchPicker` and `BlocksPicker` into two components with a common props interface. 

Then pick the widget with a lookup and throw an error if there's no match

 ```js
    const WIDGETS = { sketch: SketchWidget, blocks: BlocksWidget }

    export function ColorPicker(props) {
      const Widget = WIDGETS[props.widget]
 
      if (!Widget) throw new Error("Some kind of helpful error message. Not configured properly")
 
      // ....
      return (
        <>
          <Swatch ... />
          { displayColorPicker && <Widget ... /> }
       </>
       )
    }
```

This will make it easier to add new widgets without changing the source of the `ColorPicker` component.

Reply:
Hey @ncphillips ... I think I got this one. Not really sure. But made a new commit so check it out. =) Also .. I haven't used TS for so long so bare with me if there's something that look strange there. Also .. you have to look into what error message you want.

Removed the old class component also that was commented out

Reply:
It's a little annoying that the BlockPicker and SketchPicker uses different names on some props that are essentially the same ... Otherwise these extra components wouldn't be needed.